src/PrestaShopBundle/Controller/Admin/CommonController.php

```diff
@@ -330,6 +331,8 @@ public function renderFieldAction($formName, $formType, $fieldName, $fieldData)
      * @param string $redirectRoute
      * @param array $redirectQueryParamsToKeep
      *
+     * @AdminSecurity("is_granted(['read'], request.get('_legacy_controller'))")
```

Comment:
I dont think we need to protect this action as there is no route targeting it and it seems to be a util function, not a controller action that return a Response ?

Reply:
it doesn't seem harmful, but it dispatches a hook internally in FormFactory, so I'd rather secure it :thinking: 

there are some routes connected to this action, e.g. this one: https://github.com/PrestaShop/PrestaShop/blob/e987584b5b30cd74cf3a7b185343cfc3a0d577fe/src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/categories.yml#L8

Reply:
My bad, from github page I thought this was related `renderFieldAction` instead of `searchGridAction`. Now it makes sense 👍 .

By the way I'm wondering ... why do we have some controllers who provide their own `searchGridAction` and some controllers which use this generic action ?

Reply:
> why do we have some controllers who provide their own searchGridAction and some controllers which use this generic action ?

I don't really know, but my guess is that the generic action was introduced later and some grids already had their implementations of search action :thinking:

Reply:
Created issue https://github.com/PrestaShop/PrestaShop/issues/13448 to handle this